microcosm_fastapi/database/store.py

```diff
@@ -87,105 +89,130 @@ async def with_transaction(self, session):
             raise
 
     @postgres_metric_timing(action="create")
-    async def create(self, instance):
+    async def create(self, instance, session: Optional[AsyncSession] = None):
         """
         Create a new model instance.
         """
-        async with self.session_maker() as session:
-            async with self.with_transaction(session):
-                async with self.flushing(session):
-                    if instance.id is None:
-                        instance.id = self.new_object_id()
-                    session.add(instance)
+        if session:
+            async with self.flushing(session):
+                if instance.id is None:
+                    instance.id = self.new_object_id()
+                session.add(instance)
+
+        else:
+            async with self.session_maker() as session:
+                async with self.with_transaction(session):
+                    async with self.flushing(session):
```

Comment:
this if/else block is repeated a few times - could we factor out a content manager that takes the optional`session` and does the right thing?

Reply:
@tonybajan-globality you mean a context manager, right?

Reply:
Yes indeed

Reply:
Yep great suggestion - will implement. Thanks both

Reply:
Done